packages/jest-mock/src/index.ts

```diff
@@ -978,7 +973,7 @@ export class ModuleMocker {
       // @ts-expect-error component is a function so it has a name
       metadata.name = component.name;
       if (this.isMockFunction(component)) {
-        metadata.mockImpl = component.getMockImplementation();
+        metadata.mockImpl = component.getMockImplementation() as T;
```

Comment:
The `component` is of type `unknown`, so `isMockFunction` returns some unknown `Mock`. `mockImpl` is unhappy with that, it requires some better type. That type is `T`, it is still `unknown` and `mockImpl` is happy with that.